tests/Opus/Doi/DataCiteXmlGeneratorTest.php

```diff
@@ -139,4 +139,116 @@ public function testGetStylesheetPathConfigured()
 
         $this->assertEquals($path, $stylesheetPath);
     }
+
+    /**
+     * The DataCite-XML should not contain files, which are invisible in oai
+     * Test if both invisible files are hided
+     */
+    public function testTwoFileInformationInvisible()
+    {
+        $doc = new Opus_Document();
+        $this->addRequiredPropsToDoc($doc);
+
+        $file = New Opus_File();
+        $file->setVisibleInOai(0);
+        $file->setFileSize('0');
+        $file->setMimeType('pdf');
+        $doc->addFile($file);
+
+        $file = New Opus_File();
+        $file->setVisibleInOai(0);
+        $file->setFileSize('0');
+        $file->setMimeType('pdf');
+        $doc->addFile($file);
+
+        $generator = new Opus_Doi_DataCiteXmlGenerator();
+        $result = $generator->getXml($doc);
+
+        $this->assertNotContains('<sizes><size>0 KB</size></sizes>', $result);
+        $this->assertNotContains('<formats><format>pdf</format></formats>', $result);
+    }
+
+    /**
+     * The DataCite-XML should not contain files, which are invisible in oai
+     * Test if both visible files are shown
+     */
+    public function testTwoFileInformationVisible()
```

Comment:
Eine Nebenfunktion des Tests ist es zu prüfen, ob die Informationen richtig angezeigt werden. Daher wäre für mindestens eine Datei eine andere Größe sinnvoll. Vielleicht steht in der Ausgabe immer "0 KB". Ein anderer Dateityp wäre auch schön.